tests/test_0835-datetime-type.py

```diff
@@ -0,0 +1,63 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+from __future__ import absolute_import
+
+import pytest  # noqa: F401
+import numpy as np  # noqa: F401
+import awkward as ak  # noqa: F401
+
+
+def test_date_time():
+
+    numpy_array = np.array(
+        ["2020-07-27T10:41:11", "2019-01-01", "2020-01-01"], "datetime64[s]"
+    )
+
+    array = ak.Array(numpy_array)
+    assert (
+        str(array.type)
+        == '3 * datetime64[parameters={"__array__": "datetime64", "__datetime64_data__": "<M8[s]", "__datetime64_unit__": "1s"}]'
+    )
+    assert array.tolist() == [
+        1595846471,
+        1546300800,
+        1577836800,
```

Comment:
these are the number of `datetime64` units: seconds in this case. The units and the step are stored in ` __datetime64_unit__` parameter.

Reply:
another approach is to list them as `np.datetime64`:
```python
assert array.tolist() == [
         np.datetime64("2020-07-27T10:41:11"),
         np.datetime64("2019-01-01T00:00:00"),
         np.datetime64("2020-01-01T00:00:00"),
     ]
```

Reply:
It would be better if `tolist` makes a list of `np.datetime64` objects, but that would be automatic if `one_dimensional[0]` returns a `np.datetime64`. The `tolist` uses a recursive function defined in src/python/content.cpp that just calls `getitem_at` with appropriate `box`ing.

Reply:
yes, that's what is done in the last commit